File: cpp_solutions/2000.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int pos = -1;
        int n = word.size();
        for (int i = 0; i < n; ++i) {
            if (word[i] == ch) {
                pos = i;
                break;
            }
        }
        if (pos == -1) return word;
        reverse(word.begin(), word.begin() + pos + 1);
        return word;
    }
};

File: rust_solutions/2000.rs
impl Solution {
    pub fn reverse_prefix(word: String, ch: char) -> String {
        let index = word.find(ch);
        if index.is_none() {
            return word;
        }
        let index = index.unwrap();
        word[..index + 1].chars().rev().chain(word[index + 1..].chars()).collect()
    }
}
